### PR TITLE
Add docs on playground testnet & custom testnets

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal. This repo contains the content for https://internetcomputer.org and the ICP Developer Documentation, https://internetcomputer.org/docs/. 

If you are submitting a Pull Request for adding or changing content on the ICP Developer Documentation, please make sure that your contribution meets the following requirements:

- [ ] Follows the [developer docs style guide](style-guide.md).
- [ ] Follows the [best practices and guidelines](README.md#best-practices).
- [ ] New documentation pages include [document tags](README.md#document-tags).
- [ ] New documentation pages include [SEO keywords](README#seo-keywords).
- [ ] New documents are in the `.mdx` file format to support the previous two components. 
- [ ] New documents are registered in [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js), otherwise, it will not appear in the
  side navigation bar.
- [ ] Make sure that the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file is
  filled with new documents that you added. This way we can ensure that future Pull Requests are reviewed by the right people.

### CODE REVIEW

File: docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+git clone https://github.com/dfinity/motoko-playground
+```
+
+- #### Step 2: To create a separate pool, first use the current playground [pool and `wasm-utils` canisters as the starting point](https://github.com/dfinity/motoko-playground/tree/main/service).
+
+- #### Step 3: Then, edit the `pool/Main.mo` file to change your custom playground settings.
+
+Some settings you may want to change are:
+
+- Add [access control](https://github.com/dfinity/motoko-playground/blob/main/service/pool/Main.mo#L134) as desired, such as creating a `whitelist` of principals that are permitted to use the custom playground.
```
Comment (dfx-json) [inline docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx:53]: `whiteList` is used here and `allowList` is used in the paragraph below

File: docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    }
+  }
+}
+```
+
+To define multiple networks, use multiple definitions on different domain names:
+
+```json
+{
+  "myNetwork1": {
```
Comment (dfx-json) [inline docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx:129]: should we give these networks more real-world names? such as `staging` or `pre-production`? totally up to you. I also think using `myNetwork` illustrates the fact that these networks can be named anything

File: docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx (hunk 1 above)
Comment (jessiemongeon1) [reply docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx]: Fixed so both use `allowList`

File: docs/developer-docs/smart-contracts/deploy/overview.mdx [hunk 3]
```diff
@@ -17,26 +17,30 @@ After you have built a <GlossaryTooltip>canister</GlossaryTooltip>, you can
...
-canister to the so-called playground. You do not need cycles or tokens to deploy and
-operate your canister in the playground, but it will be automatically removed
-after 20 minutes.
+significantly lower cost than on the mainnet. Because the execution of canisters is
+fairly cheap on ICP, and ICP smart contracts can be upgraded once deployed, there is no testnet for ICP.
+Developers are encouraged to test their canister smart contracts locally or directly on the mainnet.
+
+However, the [playground](/docs/current/developer-docs/getting-started/deploy/testnet) sandbox environment can be used as a testnet-like deployment option. It does not require cycles or tokens to deploy and operate a canister, but it has limitations, and canisters will be automatically removed after 20 minutes.
+
+For most developers, the playground option can be used for most workflows, as will be demonstrated below. For more advanced developers and use cases, there are two possible options for a testnet-like environment:
```
Comment (dfx-json) [inline docs/developer-docs/smart-contracts/deploy/overview.mdx:39]: the language here might be a tad repetitive since the word `most` is being used twice

**Review by dfx-json — APPROVED** [review]
Nice work Jessie! LGTM

File: docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx (hunk 2 above)
Comment (jessiemongeon1) [reply docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx:129]: Yea I wanted to demonstrate that they can be named anything so I think they are good as is. Naming them 'staging' or 'testing' might accidentally imply they have specific parameters/usage.

File: docs/developer-docs/smart-contracts/deploy/overview.mdx (hunk 3 above)
Comment (jessiemongeon1) [reply docs/developer-docs/smart-contracts/deploy/overview.mdx]: Updated to "For the majority of developers..."

**Pull request merged**